src/http/url.rs

```diff
@@ -13,6 +13,9 @@ use crate::utils::result::ResultExt;
 
 use super::url_search_params::URLSearchParams;
 
+static DEFAULT_PORTS: &[&str] = &["21", "80", "443"];
+static DEFAULT_PROTOCOLS: &[&str] = &["ftp", "http", "https"];
```

Comment:
Using the url crate and parsing the URLs shouldn't be required. :thinking: 

Reference:
https://github.com/servo/rust-url/blob/73803fa780c8890e0b40cd01fbbd5362fbe4c9e0/url/tests/setters_tests.json#L764

Reply:
> Using the url crate and parsing the URLs shouldn't be required. 🤔

Can you expand? I don't follow

Reply:
Oh I see what you mean. I still think it's needed tho, because it should be reflected when setting `port`, `protocol` or `host`. I can work around this by parsing a fake url and extract the port from that, but it seems rather inefficient. But to your point, `filtered_port` can likely be removed from the `new` method then.

Reply:
The url crate has setters that can be used for those cases.

Reference:
https://docs.rs/url/2.5.0/url/struct.Url.html?search=set_

Reply:
That's right, but it would require us to store the url instead of the values! However this is not a bad thing as we can refactor the URL module to instead hold a `url` value and just call methods of that instead of extracting the values from a single parse. Would you be interested in taking this on? PR Welcome 👍

Reply:
I will work on it as soon as I have some time.